### DISCUSSION TITLE
Expand table on column resize

### DISCUSSION THREAD

**Discussion by HaveSpacesuit** [iTwin/iTwinUI-react#822] [Ideas]
It is difficult to expand columns to a wide width. Currently it looks like when a column is resized, the next column loses or gains width based on the first column resize. The overall table width stays the same.

![Column resize static](https://user-images.githubusercontent.com/15036346/189432427-19eb901a-f641-4cf3-84ee-48a3ed4f0778.gif)

It would be nice to expand/shrink the entire table width when a column is resized. If I make column A bigger, I likely do not want to make column B smaller.

This should work in situations when the table has horizontal scroll and does not have horizontal scroll - adding or removing the scroll as needed.

Here is a sandbox showing that it is not easy to expand a column past the point where the next column is at minimum width. There are examples with and without horizontal scroll. (I usually have the last column width set to undefined in order to allow the table to fill horizontal space if the columns do not all fit.)

https://codesandbox.io/s/itwinui-react-minimal-example-forked-7wizuj?file=/src/columns.ts

![Column width scroll](https://user-images.githubusercontent.com/15036346/189432701-0fe14f6b-6464-413e-9dc2-986e561dd567.gif)

**Comment by HaveSpacesuit** [comment]
**Update:**

I was able to hack the react-table state to make it perform close to how I want, but I'm not proud of myself. It requires manually tweaking react-table's state in a way that doesn't seem intended.

https://codesandbox.io/s/itwinui-react-minimal-example-forked-3ydduv?file=/src/App.tsx

```ts
const stateReducer = useCallback(
  (newState: TableState, action: ActionType) => {
    if (action.type == "columnResizing") {
      const innocentColumn = newState.columnResizing.nextHeaderIdWidths;
      if (innocentColumn) {
        const [innocentName, innocentWidth] = innocentColumn[0];
        newState.columnResizing.columnWidths[innocentName] = innocentWidth;
      }
    }
    return newState;
  },
  []
);
```

It also doesn't work great, as it stops expanding the column after a point...

![Column expansion stops](https://user-images.githubusercontent.com/15036346/189541007-4dd005ae-9139-4464-adbd-5c94d18f8aaf.gif)

**Comment by bentleyvk** [comment]
Hi, it works as designed (and I spent a lot of time making it to work 😁), we copied this behavior from Word tables because if there is a table without horizontal scroll and you don't want it, it is really cool that resize only affects closest columns. I think I have seen one table that has a flag for resize type so maybe we could do something similar, need to discuss with the team.

**Reply by bentleyvk** [reply]
@HaveSpacesuit we added `columnResizeMode` to the Table in [v1.47.0](https://github.com/iTwin/iTwinUI-react/releases/tag/v1.47.0)
